### PR TITLE
test: isolate ssh_import_id to its own class and tolerate GH rate limits

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message
```
test: isolate ssh_import_id to its own class and tolerate GH rate limits

ssh_import_id was in the shared USER_DATA used by both TestCombined and
TestCombinedNoCI, so a single GitHub REST API 403 rate limit cascaded
into three failures (test_ssh_import_id, test_no_problems,
test_deprecated_message via verify_clean_boot).

Move ssh_import_id into a dedicated TestSshImportId class with its own
user-data to limit the blast radius of a GitHub rate limiting to one test.
Remove from the shared USER_DATA and from test_no_problems.

The isolated test detects the rate-limit signature
('GitHub REST API rate-limited' / 'status_code=403') in
cloud-init-output.log and pytest.skips instead of hard-failing.
```

## Additional Context
[Intermittent failed  ssh-import-id workflows on scheduled GH workfow runs](https://github.com/canonical/cloud-init/actions/runs/30314158711/job/90136041230)

## Test Steps
```
CLOUD_INIT_CLOUD_INIT_SOURCE=ppa:cloud-init-dev/daily CLOUD_INIT_PLATFORM=lxd_vm CLOUD_INIT_OS_IMAGE=stonking tox -e integration-tests -- tests/integration_tests/modules/test_combined.py
```
## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/integration_tests/modules/test_combined.py [hunk 1]
```diff
@@ -661,21 +657,42 @@ def test_networkd_wait_online(self, class_client: IntegrationInstance):
...
+        output_log = ""
+        try:
+            output_log = class_client.read_from_file(
+                "/var/log/cloud-init-output.log"
+            )
+        except IOError:
+            pass
+        if (
+            "GitHub REST API rate-limited" in output_log
+            or "status_code=403" in output_log
```
Comment (holmanb) [inline tests/integration_tests/modules/test_combined.py]: It seems like this could cause the test to be skipped for reasons besides rate-limiting.

File: tests/integration_tests/modules/test_combined.py [hunk 2]
```diff
@@ -661,21 +657,42 @@ def test_networkd_wait_online(self, class_client: IntegrationInstance):
...
         """
-        client = class_client
-        ssh_output = client.read_from_file("/home/ubuntu/.ssh/authorized_keys")
+        output_log = ""
+        try:
+            output_log = class_client.read_from_file(
+                "/var/log/cloud-init-output.log"
+            )
+        except IOError:
+            pass
```
Comment (holmanb) [inline tests/integration_tests/modules/test_combined.py]: This is an odd exception to ignore. I understand that the test could still pass in this case, but that seems like an edge case which complicates the code - not an expected error. Have we seen this happen before?

Also, IOError is just an alias for OSError. I would suggest not using this exception in the future for consistency.

**Review by holmanb — CHANGES_REQUESTED** [review]
Thanks for this. There is a merge conflict.

File: tests/integration_tests/modules/test_combined.py (hunk 2 above)
Comment (blackboxsw) [reply tests/integration_tests/modules/test_combined.py]: I can't seem to find a test case where this error was raised anymore. I'll drop it for the time being to simplify this changeset. We can add the OSError handling if we encounter it again in the future.

File: tests/integration_tests/modules/test_combined.py (hunk 1 above)
Comment (blackboxsw) [reply tests/integration_tests/modules/test_combined.py]: I've also backed out the status_code=403 for the moment as that could more generically match a number of things. If we find a case where `Github REST API rate-limited` is not sufficient, we can build in a more specific catch for status_code=403 in context with the remainder of the unique failure message.

**Review by holmanb — APPROVED** [review]
Thanks!
